godot-core/src/builtin/dictionary.rs

```diff
@@ -320,6 +311,210 @@ impl Share for Dictionary {
     }
 }
 
+struct DictionaryIter<'a> {
+    last_key: Option<Variant>,
+    dictionary: &'a Dictionary,
+    is_first: bool,
+}
+
+impl<'a> DictionaryIter<'a> {
+    fn new(dictionary: &'a Dictionary) -> Self {
+        Self {
+            last_key: None,
+            dictionary,
+            is_first: true,
+        }
+    }
+
+    fn call_init(dictionary: &Dictionary) -> Option<Variant> {
+        // SAFETY:
+        // `dictionary` is a valid `Dictionary` since we have a reference to it,
+        //    so this will call the implementation for dictionaries.
+        // `variant` is an initialized and valid `Variant`.
+        let variant: Variant = Variant::nil();
+        unsafe { Self::call_iter_fn(interface_fn!(variant_iter_init), dictionary, variant) }
+    }
+
+    fn call_next(dictionary: &Dictionary, last_key: Variant) -> Option<Variant> {
+        // SAFETY:
+        // `dictionary` is a valid `Dictionary` since we have a reference to it,
+        //    so this will call the implementation for dictionaries.
+        // `last_key` is an initialized and valid `Variant`, since we own a copy of it.
+        unsafe { Self::call_iter_fn(interface_fn!(variant_iter_next), dictionary, last_key) }
+    }
+
+    /// SAFETY:
+    /// `iter_fn` must point to a valid function that interprets the parameters according to their type specification.
```

Comment:
For API documentation (`///`), the convention is to add a new section (h1 as `# Safety`).
The `// SAFETY:` syntax is for code comments (`//`).

Clippy _should_ check the first.
```suggestion
    /// # Safety
    /// `iter_fn` must point to a valid function that interprets the parameters according to their type specification.
```